net: latmon: Fix net API use

aescolar opened the pull request:
In d45cd6716bbab3a805e3a5fd461934f0dcdc13e5 the mayority of the Zephyr networking code was changed to use the Zephyr native net_/zsock_ prefixed types, but some symbols were missed. 
Let's change them.

Without these fixes/changes the code still builds in most cases as we are by now setting CONFIG_NET_NAMESPACE_COMPAT_MODE.
But when this is not set, things will fail to build.

And change the respective sample app to use native networking API directly in the one call it wasn't.

aescolar committed:
net: latmon: Fix net API use

In d45cd6716bbab3a805e3a5fd461934f0dcdc13e5 the mayority of the Zephyr
networking code was changed to use the Zephyr native net_/zsock_
prefixed types, but some symbols were forgotten.
Let's change them.

Without these fixes/changes the code still builds in most cases as we are
by now setting CONFIG_NET_NAMESPACE_COMPAT_MODE.
But when this is not set, things will fail to build.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples: net: latmon: Use native networking API directly

This subsystem does not use the POSIX_API, and the sample only used one
call to the POSIX API to close a socket instead of using the native
networking API.
Let's just use the native networking API, so we avoid pulling-in more
dependencies.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

jukkar reviewed (commented):
LGTM. One reason why the library code was not noticed to contain POSIX symbols is that `tests/net/all/prj.conf` is missing enabling LATMON i.e., `CONFIG_NET_LATMON=y`
While at it, could you add a commit to this PR enabling that?

aescolar committed:
tests: net: all: Also build latmon

Also build the latency monitor support code.

As a sideeffect, we run out of bits in the kernel objects per thread
permissions tracking field. So let's increase it to 6 bytes.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar commented:
> tests/net/all/prj.conf is missing enabling LATMON i.e., CONFIG_NET_LATMON=y
While at it, could you add a commit to this PR enabling that?

Done :)

aescolar committed:
modules: openthread: Fix network namespace API usage

bcec3bd04ba38e65008f81d867f3b046a9da637f introduced some of the
non-native API names back into this module. Let's fix it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar commented:
The last push was just a rebase + an extra commit with a fix for the issue which was just introduced in main in the openthread integration which also triggered in this PR.

aescolar commented:
I labelled the PR trivial so we can skip the few last minutes of the 48h if we want to. Any release engineer can merge it now.
The changes since the last batch of approvals was just the hotfix for the openthread module integration issue in main which also triggered in this PR. This last fix is conceptually trivial.
Before the PR had been approved by both networking maintainers.
